src/bearbyWallet/BearbyAccount.ts

```diff
@@ -69,30 +68,22 @@ export class BearbyAccount implements IAccount {
   public async balance(): Promise<IAccountBalanceResponse> {
     // TODO: check if we need to connect every time
     await this.connect();
-    // Not available on bearby. we have to manually call the api
-    const body = {
-      jsonrpc: '2.0',
-      method: 'get_addresses',
-      params: [[this._address]],
-      id: 0,
-    };
-
-    // get node url: This is a temporary solution. We should get the balance from the provider
-    const nodeUrl = await getNodesUrl();
 
-    const addressInfos = await postRequest<BalanceResponse>(nodeUrl, body);
+    const res = await web3.massa.getAddresses(this._address);
 
-    if (addressInfos.isError || addressInfos.error) {
-      throw addressInfos.error.message;
+    if (!res && !res.result) {
+      throw new Error(res.error.message);
     }
 
+    // TODO: check if we need to cast to AddressInfo. Fix to do on bearby.js
+    const addressInfo = res?.result[0] as AddressInfo;
```

Comment:
res.result[0]

Reply:
if i remove as AddressInfo, final_balance and candidate_balance are not reconized

Reply:
If you were only talking about the "?", fixed it 🙏